### PR TITLE
add new node type selector to `databricks_node_type` data source

### PR DESCRIPTION
**Pull request by alexott**

### CODE REVIEW

**Review by nfx — APPROVED** [review]
please don't forget to update SDK

File: clusters/data_node_type.go [hunk 1]
```diff
@@ -136,6 +137,9 @@ func (a ClustersAPI) GetSmallestNodeType(r NodeTypeRequest) string {
 		if r.VCPU {
 			return "vcpu-worker"
 		}
+		if r.Fleet {
+			return "m-fleet.xlarge"
```
Comment (abagshaw) [inline clusters/data_node_type.go:141]: Let's use `md-fleet.xlarge` as that supports photon.
```suggestion
			return "md-fleet.xlarge"
```

File: clusters/data_node_type.go [hunk 2]
```diff
@@ -150,6 +154,12 @@ func (a ClustersAPI) GetSmallestNodeType(r NodeTypeRequest) string {
 		if !r.VCPU && strings.HasPrefix(nt.NodeTypeID, "vcpu") {
 			continue
 		}
+		if r.Fleet && !strings.Contains(nt.NodeTypeID, "-fleet.") {
+			continue
+		}
+		if !r.Fleet && strings.Contains(nt.NodeTypeID, "-fleet.") {
+			continue
+		}
```
Comment (abagshaw) [inline clusters/data_node_type.go:162]: nit:
```suggestion
		if r.Fleet != strings.Contains(nt.NodeTypeID, "-fleet.") {
			continue
		}
```

**Review by abagshaw — APPROVED** [review]
LGTM! Minor comments

**Review by nfx — APPROVED** [review]

**Pull request merged**